api_app/core/serializers.py

```diff
@@ -163,6 +164,31 @@ def _md5_config_file(cls) -> str:
             md5hash = hashlib.md5(buffer).hexdigest()
         return md5hash
 
+    @classmethod
+    def _complete_config(
+        cls, config_dict: dict, plugin_name: str, visited: set
+    ) -> dict:
+        """
+        Completes config by parsing extends configs
+        """
+        if plugin_name in visited:
+            raise RuntimeError(f"Circular dependency detected in {cls} config")
+        visited.add(plugin_name)
+        if plugin_name not in config_dict:
+            raise RuntimeError(
+                f"Plugin {plugin_name} not found in {cls} config "
+                "but referenced in extends"
+            )
+        if "extends" in config_dict[plugin_name]:
```

Comment:
shouldn't we add this as an optional attribute in the serializer?

Reply:
done